core/src/plugins/filed/python/libcloud/BareosFdPluginLibcloud.py

```diff
@@ -357,7 +363,10 @@ def create_file(self, restorepkt):
             100, "create_file() entry point in Python called with %s\n" % (restorepkt)
         )
         FNAME = FilenameConverter.BackupToBucket(restorepkt.ofname)
-        dirname = os.path.dirname(FNAME)
+        if version_info.major < 3:
+            dirname = os.path.dirname(FNAME)
+        else:
+            dirname = os.path.dirname(FNAME).encode("utf-8")
```

Comment:
This logic appears inverted with the changes on line 322-325 above. Is that intentional?

Reply:
Good point, but yes, it is intentional because lines 366-369 are for restore, while 322-325 are for backup.